test/TestUtilities/ApproxArray.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SFML_TESTUTILITIES_APPROXARRAY_HPP
+#define SFML_TESTUTILITIES_APPROXARRAY_HPP
+
+#include <doctest.h>
+#include <array>
+
+namespace std
+{
+    template <typename T, std::size_t size>
+    std::ostream& operator<<(std::ostream& out, const std::array<T, size>& array)
+    {
+        if (size == 0)
+            return out << "{}";
+
+        out << "{ ";
+        for (size_t i = 0; i + 1 < array.size(); ++i)
+            out << array[i] << ", ";
+        out << array.back() << " }";
+
+        return out;
+    }
+}
+
+namespace sf::Testing
+{
+    template <typename T, std::size_t size>
+    class ApproxArray
+    {
+    public:
+        ApproxArray(const std::array<T, size>& values)
+        : values{values}
+        {}
+
+        const std::array<T, size>& values;
+    };
+
+    template <typename T, std::size_t size>
+    bool operator==(const ApproxArray<T, size>& approx, const std::array<T, size>& array)
+    {
+        for (std::size_t i = 0; i < size; i++)
+            if (approx.values[i] != doctest::Approx(array[i]))
+                return false;
+
+        return true;
+    }
+
+    template <typename T, std::size_t size>
+    bool operator==(const std::array<T, size>& array, const ApproxArray<T, size>& approx)
+    {
+        return approx == array;
+    }
+
+    template <typename T, std::size_t size>
+    bool operator!=(const ApproxArray<T, size>& approx, const std::array<T, size>& array)
+    {
+        return !(approx == array);
+    }
+
+    template <typename T, std::size_t size>
+    bool operator!=(const std::array<T, size>& array, const ApproxArray<T, size>& approx)
+    {
+        return !(approx == array);
+    }
```

Comment:
What's the use case for != in this context?

Reply:
I agree, we could settle for the convention `exact == approx` and `exact != approx` everywhere, this is not exposed to user code.

Reply:
Right, we only use `exact == approx`, so I will remove other comparisons.